frontend/src/Sidebar.res

```diff
@@ -157,15 +157,15 @@ module WorkersSelect = {
       workers->Belt.Array.getIndexBy(w => worker == Some((w.worker, w.docker_image)))
     }
 
-    React.useEffect1(() => {
-      switch idx_opt {
-      | None =>
+    React.useEffect2(() => {
+      switch (idx_opt, workers->Belt.Array.length > 0) {
+      | (None, true) =>
```

Comment:
Fun trick: pattern matching can use an [`if` clause](https://rescript-lang.org/docs/manual/latest/pattern-matching-destructuring#if-clause) to simplify this sort of code (in ocaml the keyword is `when`)

Reply:
Yes, thanks for the suggestion! I am aware of this and have used it in the past, but I'm still not super comfortable with OCaml idioms/tricks to use them consistently. Pushed a commit to change this.